cpp/src/io/json/read_json.cu

```diff
@@ -41,6 +42,56 @@ namespace cudf::io::json::detail {
 
 namespace {
 
+class compressed_host_buffer_source final : public datasource {
```

Comment:
interesting approach
could we wrap a datasource instead of passing a host buffer?

Reply:
Yes, that was the other approach I was thinking of. We can have a owning buffer as a member of the `compressed_host..`  class in that case.